WebApp-OpenIDConnect-DotNet/Controllers/SessionController.cs

```diff
@@ -21,18 +22,28 @@ public SessionController(IOptions<AzureAdB2COptions> b2cOptions)
         public AzureAdB2COptions AzureAdB2COptions { get; set; }
 
         [HttpGet]
-        public IActionResult SignIn()
+        public async Task SignIn()
         {
-            return Challenge(
-                new AuthenticationProperties { RedirectUri = "/" }, OpenIdConnectDefaults.AuthenticationScheme);
+            await HttpContext.Authentication.ChallengeAsync(
+                OpenIdConnectDefaults.AuthenticationScheme, new AuthenticationProperties { RedirectUri = "/" });
         }
 
         [HttpGet]
-        public IActionResult ResetPassword()
+        public async Task ResetPassword()
         {
-            var properties = new AuthenticationProperties() { RedirectUri = "/" };
+            var properties = new AuthenticationProperties() { RedirectUri = "/"  };
             properties.Items[AzureAdB2COptions.PolicyAuthenticationProperty] = AzureAdB2COptions.ResetPasswordPolicyId;
-            return Challenge(properties, OpenIdConnectDefaults.AuthenticationScheme);
+            await HttpContext.Authentication.ChallengeAsync(
```

Comment:
@tratcher @haok The ChallengeResult returned by the Challenge helper method just calls ChallengeAsync on the authentication manager under the covers, so I think this is technically ok, but it's certainly not as nice. Why is specifying the ChallengeBehavior explicitly as Unauthorized required instead of leaving the behavior as automatic? I was able to reset my password without this. Where do you see this fail?